### DISCUSSION TITLE
partials, header levels, and TOC

### DISCUSSION THREAD

**Discussion by ryandeussing** [markdoc/markdoc#272] [Q&A] [answered]
I'm evaluating MarkDoc as a replacement for AsciiDoc, which offers a `leveloffset` attribute for its `include` directive. In Markdoc terms, this allows a partial be included with headings that are adjusted, relatively, based on the context of the parent document.

What might be the best approach to achieving this in Markdoc? Perhaps replacing the `partial` tag with one that looks for something like an `offset` variable and performs a transform?

~And given a document that is built with partials, how might I run a table of contents function on output that includes the partial/s and the/ir headers?~ (this appears to Just Work®️)

Thanks!

**Comment by rpaul-stripe** [comment, marked as answer]
You could use an AST transform, but I think the easiest way to do this is to use the `variables` attribute on the partial tag to pass a `leveloffset` variable to the partial's children and then override the heading node so that it adds the value of `leveloffset` to the actual level when it detects that the variable is defined:

```
{% partial file="partial.md" variables={leveloffset: 1} /%}
```

```
heading: {
  ...Markdoc.nodes.heading,
  transform(node, config) {
    const offset = config.variables?.leveloffset ?? 0;
    const level = offset + node.attributes.level;

    return new Markdoc.Tag(
      `h${level}`,
      node.transformAttributes(config),
      node.transformChildren(config)
    );
  },
},
```

**Reply by ryandeussing** [reply]
Thanks, much appreciated!
